### PR TITLE
MULE-13575: File connector tests using locks fail on Windows

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: src/main/java/org/mule/extension/ftp/internal/command/FtpReadCommand.java [hunk 1]
```diff
@@ -55,22 +55,18 @@ public FtpReadCommand(FtpFileSystem fileSystem, FTPClient client) {
...
+      payload = ClassicFtpInputStream.newInstance((FtpConnector) config, attributes, pathLock);
       return Result.<InputStream, FileAttributes>builder().output(payload)
           .mediaType(fileSystem.getFileMessageMediaType(attributes))
           .attributes(attributes).build();
-    } catch (ConnectionException e) {
-      throw exception("Could not obtain connection to fetch file " + path, e);
+    } catch (Exception e) {
+      pathLock.release();
+      closeQuietly(payload);
+      throw exception("Could not obtain fetch file " + path, e);
```
Comment (alegmarra) [inline src/main/java/org/mule/extension/ftp/internal/command/FtpReadCommand.java]: error?

File: src/main/java/org/mule/extension/ftp/internal/command/FtpWriteCommand.java [hunk 2]
```diff
@@ -49,33 +54,39 @@ public void write(String filePath, InputStream content, FileWriteMode mode, bool
...
-      IOUtils.copy(content, outputStream);
-      LOGGER.debug("Successfully wrote to path {}", normalizedPath);
-    } catch (Exception e) {
-      throw exception(format("Exception was found writing to file '%s'", normalizedPath), e);
+      String normalizedPath = normalizePath(path);
+      try (OutputStream outputStream = getOutputStream(normalizedPath, mode)) {
+        IOUtils.copy(content, outputStream);
+        LOGGER.debug("Successfully wrote to path {}", normalizedPath);
+      } catch (Exception e) {
+        throw exception(format("Exception was found writing to file '%s'", normalizedPath), e);
```
Comment (alegmarra) [inline src/main/java/org/mule/extension/ftp/internal/command/FtpWriteCommand.java]: write error?

Comment (marianogonzalez) [reply src/main/java/org/mule/extension/ftp/internal/command/FtpWriteCommand.java]: not

File: src/main/java/org/mule/extension/ftp/internal/command/FtpReadCommand.java (hunk 1 above)
Comment (marianogonzalez) [reply src/main/java/org/mule/extension/ftp/internal/command/FtpReadCommand.java]: not

**Pull request merged**